webui/react/src/components/UPlot/UPlotChart.tsx

```diff
@@ -26,141 +25,208 @@ interface ScaleZoomData {
+  if (prev.scales?.y?.distr !== next.scales?.y?.distr) return true;
+
+  const someSeriesHasChanged =
+    chart.series.some((chartSerie, seriesIdx) => {
+      const nextSerie = next.series?.[seriesIdx];
+      const prevSerie = prev.series?.[seriesIdx];
+      return (
+        (nextSerie?.show != null && chartSerie?.show !== nextSerie?.show)
+        || (prevSerie?.show != null && prevSerie?.show !== nextSerie?.show)
+        || (nextSerie?.label != null && chartSerie?.label !== nextSerie?.label)
+        || (nextSerie?.fill != null && chartSerie?.fill !== nextSerie?.fill)
+      );
+    });
+  if(someSeriesHasChanged) return true;
+
+  return false;
+};
+
+const getNormalizedData = (
+  data: AlignedData | FacetedData | undefined,
+  mode: uPlot.Mode | undefined,
+) => {
+  if (!data || data.length < 2) return [ false, undefined ];
+
+  // Is the chart aligned (eg. linear) or faceted (eg. scatter plot)?
+  if (mode === 2) {
+    return [ true, data as AlignedData ];
+  } else {
+    // Figure out the lowest sized series data.
+    const chartData = data as AlignedData;
+    const minDataLength = chartData.reduce((acc: number, series: UPlotData[]) => {
+      return Math.min(acc, series.length);
+    }, Number.MAX_SAFE_INTEGER);
+
+    // Making sure the X series and all the other series data are the same length;
+    const trimmedData = chartData.map((series) => series.slice(0, minDataLength));
+
+    // Checking to make sure the X series has some data.
+    const hasXValues = (trimmedData?.[0]?.length ?? 0) !== 0;
+
+    return [ hasXValues, trimmedData as unknown as AlignedData ];
+  }
+};
 const SCROLL_THROTTLE_TIME = 500;
 
-const UPlotChart: React.FC<Props> = ({ data, focusIndex, options, style }: Props) => {
+const UPlotChart: React.FC<Props> = ({
+  data,
+  focusIndex,
+  options,
+  style,
+  noDataMessage,
+}: Props) => {
   const chartRef = useRef<uPlot>();
   const chartDivRef = useRef<HTMLDivElement>(null);
-  const scalesRef = useRef<Record<RecordKey, uPlot.Scale>>();
   const scalesZoomData = useRef<Record<string, ScaleZoomData>>({});
-  const isZoomed = useRef<boolean>(false);
-  const mousePosition = useRef<[number, number]>();
 
-  const [ hasData, normalizedData ] = useMemo(() => {
-    if (!data || data.length < 2) return [ false, undefined ];
-
-    // Is the chart aligned (eg. linear) or faceted (eg. scatter plot)?
-    if (options?.mode === 2) {
-      return [ true, data as AlignedData ];
-    } else {
-      // Figure out the lowest sized series data.
-      const chartData = data as AlignedData;
-      const minDataLength = chartData.reduce((acc: number, series: UPlotData[]) => {
-        return Math.min(acc, series.length);
-      }, Number.MAX_SAFE_INTEGER);
+  const getAugmentedOptions = (options: Partial<uPlot.Options> | undefined): uPlot.Options =>
+    uPlot.assign(
+      {
+        hooks: {
+          setScale: [
+            (uPlot: uPlot, scaleKey: string) => {
+              // fires *after* scale change
```

Comment:
**super nit:** idk if there is a eslint rule for this, but for one liner comments to read like a sentence. Capitalized first word and a period at the end.

Reply:
will change this